test/models/apple/publisher_test.rb

```diff
@@ -611,5 +611,101 @@
+
+      upload_media_mock = Minitest::Mock.new
+      # Called with empty array
+      upload_media_mock.expect(:call, nil, [[]])
+
+      process_and_deliver_mock = Minitest::Mock.new
+      # Called with empty array
+      process_and_deliver_mock.expect(:call, nil, [[]])
+
+      apple_publisher.stub(:upload_media!, upload_media_mock) do
+        apple_publisher.stub(:process_and_deliver!, process_and_deliver_mock) do
+          apple_publisher.stub(:publish_drafting!, publish_mock) do
+            apple_publisher.deliver_and_publish!([episode])
+          end
+        end
+      end
+
+      assert publish_mock.verify
+      assert upload_media_mock.verify
+      assert process_and_deliver_mock.verify
+    end
+
+    it "calls delivery and publishing for episodes needing delivery" do
+      episode.feeder_episode.apple_mark_as_uploaded!
+      episode.feeder_episode.apple_mark_as_not_delivered!
+
+      # Both delivery and publishing should be called
+      publish_mock = Minitest::Mock.new
+      publish_mock.expect(:call, nil, [[episode]])
+
+      delivery_mock = Minitest::Mock.new
+      delivery_mock.expect(:call, nil, [[episode]])
+
+      apple_publisher.stub(:upload_media!, ->(*) {}) do
+        apple_publisher.stub(:process_and_deliver!, delivery_mock) do
+          apple_publisher.stub(:publish_drafting!, publish_mock) do
+            apple_publisher.deliver_and_publish!([episode])
+          end
+        end
+      end
+
+      assert publish_mock.verify
+      assert delivery_mock.verify
+    end
+  end
+
+  describe "#process_and_deliver!" do
+    let(:episode) { build(:uploaded_apple_episode, show: apple_publisher.show) }
+
+    it "marks episodes as delivered and resets asset wait, but does not publish" do
+      # Setup mocks to verify method calls
+      increment_mock = Minitest::Mock.new
+      increment_mock.expect(:call, nil, [[episode]])
+
+      wait_upload_mock = Minitest::Mock.new
+      wait_upload_mock.expect(:call, nil, [[episode]])
+
+      wait_asset_mock = Minitest::Mock.new
+      wait_asset_mock.expect(:call, nil, [[episode]])
+
+      mark_delivered_mock = Minitest::Mock.new
+      mark_delivered_mock.expect(:call, nil, [[episode]])
+
+      reset_mock = Minitest::Mock.new
+      reset_mock.expect(:call, nil, [[episode]])
+
+      # Set up method stubs
+      apple_publisher.stub(:increment_asset_wait!, increment_mock) do
+        apple_publisher.stub(:wait_for_upload_processing, wait_upload_mock) do
+          apple_publisher.stub(:wait_for_asset_state, wait_asset_mock) do
+            apple_publisher.stub(:mark_as_delivered!, mark_delivered_mock) do
+              apple_publisher.stub(:reset_asset_wait!, reset_mock) do
+                # This should raise an error if publish_drafting! is called
+                apple_publisher.stub(:publish_drafting!, ->(*) { raise "publish_drafting! should not be called!" }) do
+                  apple_publisher.process_and_deliver!([episode])
```

Comment:
so many stubs! definitely not easy to create this test condition, great to see it.